File: Eunleelee/2024-02/2월 1일/달리기_경주.py

def solution(players, callings):

    player_rank = {}
    for i, player in enumerate(players):
        player_rank[player] = i
    
    for name in callings:
        idx = player_rank.get(name)
        front_player = players[idx - 1]
        
        
        player_rank[name] = idx - 1
        player_rank[front_player] = idx
        
        players[idx - 1] = name      
        players[idx] = front_player

    return players

File: yht0827/2월/달리기 경주.java
import java.util.*;

class Solution {
	public String[] solution(String[] players, String[] callings) {
		int numOfPlayers = players.length;
		Map<String, Integer> ranking = new HashMap<>();

		for (int i = 0; i < numOfPlayers; i++) {
			ranking.put(players[i], i);
		}

		for (String player : callings) {
			int playerRanking = ranking.get(player);

			String frontPlayer = players[playerRanking - 1];

			ranking.replace(frontPlayer, playerRanking);
			players[playerRanking] = frontPlayer;

			ranking.replace(player, playerRanking - 1);
			players[playerRanking - 1] = player;
		}

		return players;
	}
}